posts/jenkins-vs-aws-codepipeline.md

```diff
@@ -0,0 +1,49 @@
+---
+authors:
+- Jonathan Scharf
+tags:
+- Jenkins
+- CodePipeline
+- AWS
+- CICD
+date: 2021-08-24T12:21:50.000Z
+title: "Jenkins vs AWS CodePipeline"
+image: 
+---
+
+If you have worked with CI/CD, chances are you have used Jenkins,GitLab, or GitHub Actions, but one service I don’t often hear folks talk about is AWS CodePipeline. In this blog, we will make some comparisons between Jenkins and AWS CodePipeline, and see if maybe it’s time for your projects to make the switch.
+
+
+# What is Jenkins?
+
+Jenkins seems like the place to start. As of writing this blog, Jenkins is by far the market leader for CI/CD pipelines, and has the added benefit of being open source. Boasting over 1,800 plugins, Jenkins has been able to integrate with so many different tools and services with little additional work. Jenkins also easily integrates with all of the major public cloud providers, and many developer tools. Jenkins’ other main benefit is its price, free. While there are companies that offer Jenkins as a SaaS solution, anyone can download and start using Jenkins. This of course comes with the downside of needing to maintain everything about your Jenkins environment, which at scale can become difficult.
+
+
+# What is AWS CodePipeline?
+
+AWS CodePipeline is a fully managed CI/CD service, and is actually the orchestration of a few other AWS services including CodeCommit, CodeBuild, and CodeDeploy. While being an AWS service you can use CodePipeline to deploy to other servers running the agent. Since these are all managed services, there is no underlying infrastructure you will have to maintain. But managed services come with a cost. There is a cost for having an active pipeline, as well as the compute time required for build and some deployment operations. 
```

Comment:
```suggestion
AWS CodePipeline is a fully managed CI/CD service, and is actually the orchestration of a few other AWS services including CodeCommit, CodeBuild, and CodeDeploy. Because it is an AWS service, you can use CodePipeline to deploy to other servers running the agent. Since these are all managed services, there is no underlying infrastructure you will have to maintain. But managed services come with a cost. There is a cost for having an active pipeline, as well as the compute time required for building and some deployment operations. 
```